code/game/objects/items/stacks/stack.dm

```diff
@@ -268,14 +277,17 @@
 
 /obj/item/stack/attack_hand(mob/user as mob)
 	if (user.get_inactive_hand() == src)
+		var/thermal_mass_used = thermal_mass/amount
 		var/obj/item/stack/F = new src.type( user, amount=1)
 		F.copy_evidences(src)
 		F.material_type = material_type
 		user.put_in_hands(F)
 		src.add_fingerprint(user)
 		F.add_fingerprint(user)
 		F.transfer_data_from(src,1)
+		F.thermal_mass = thermal_mass_used
 		use(1)
+		thermal_mass -= thermal_mass_used
```

Comment:
Shouldn't `use()` itself update `thermal_mass`? It seems very funky that it doesn't